xugrid/core/utils.py

```diff
@@ -63,7 +63,7 @@ def partition(grids: list[UgridType]):
     parts = []
```

Comment:
This stackoverflow implementation seems more complicated than necessary.

Why partition it, then take the first of each partition?

```python
def unique_grids(grids: list[UgridType]) -> list[UgridType]:
    uniques: list[UgridType] = []
    for grid in grids:
        for unique in uniques:
            if grid.equals(unique):
                break
        else:
            uniques.append(grid)
    return uniques
```

This also avoids a potential minor confusion, since there's also the topological partitioning, which is very different from the partitions created here.

Reply:
Ah good suggestion, changed it!